File: problems/1710-maximum-units-on-a-truck/solution.cpp
class Solution {
public:
    int maximumUnits(vector<vector<int>>& boxTypes, int truckSize) {
        sort(boxTypes.begin(), boxTypes.end(), [](const vector<int>& a, const vector<int>& b) {
            return a[1] > b[1];
        });

        int totalUnits = 0;
        for (const auto& boxType : boxTypes) {
            int numBoxes = boxType[0];
            int unitsPerBox = boxType[1];

            if (truckSize >= numBoxes) {
                totalUnits += numBoxes * unitsPerBox;
                truckSize -= numBoxes;
            } else {
                totalUnits += truckSize * unitsPerBox;
                truckSize = 0;
                break;
            }
        }

        return totalUnits;
    }
};

File: problems/1710-maximum-units-on-a-truck/solution.java
import java.util.Arrays;

class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        Arrays.sort(boxTypes, (a, b) -> b[1] - a[1]);
        int totalUnits = 0;
        for (int[] boxType : boxTypes) {
            int numberOfBoxes = boxType[0];
            int numberOfUnitsPerBox = boxType[1];
            if (truckSize >= numberOfBoxes) {
                totalUnits += numberOfBoxes * numberOfUnitsPerBox;
                truckSize -= numberOfBoxes;
            } else {
                totalUnits += truckSize * numberOfUnitsPerBox;
                truckSize = 0;
                break;
            }
        }
        return totalUnits;
    }
}